labrad/node/__init__.py

```diff
@@ -152,168 +157,162 @@ def timeout(self):
+            twisted/internet/dumbwin32proc.py:Process
 
-    @inlineCallbacks
-    def _start(self):
-        if self.started:
-            return
-        print("starting '%s'..." % self.name)
-        print("path:", self.path)
-        print("args:", self.args)
-        self.starting = True
-        self.startup = defer.Deferred()
-        self.emitMessage('server_starting')
-        # This looks crazy.  On Unix, spawnProcess calls fork() and then
-        # execvpe which expects an executable name and a series of arguments,
-        # and tries to resolve the executable using PATH.  On windows,
-        # it calls createProcess, which expects a command string and an
-        # optional executable.  If the executable argument is present,
-        # it must be an absolute path including .exe suffix.  If executable
-        # is not present, the first part of the command string is used,
-        # and invokes the %PATH% and extension search.
-        #
-        # Twisted spawnProcess only supports posix (including Mac) and
-        # win32, so we just abandon anything else.
-        # Relevant code is in twisted/internet/process.py:_BaseProcess and
-        # twisted/internet/dumbwin32proc.py:Process. The twisted docs for
-        # spawnProcess say that the full executable path is required, but
-        # this is not in fact the case. See:
-        # http://twistedmatrix.com/documents/current/api/twisted.internet.interfaces.IReactorProcess.spawnProcess.html
+        Note that the twisted docs for spawnProcess say that the full executable
+        path is required, but this is not in fact the case. See:
+
+            http://twistedmatrix.com/documents/current/api/twisted.internet.interfaces.IReactorProcess.spawnProcess.html
+        """
         if platformType == 'posix':
-            executable = self.args[0]
+            return self.args[0]
         elif platformType == 'win32':
-            executable = None
+            return None
         else:
             raise RuntimeError("Unsupported platform %s" % platformType)
 
-        self.proc = reactor.spawnProcess(self, executable, self.args,
+    def set_status(self, value):
+        """Transition to the given status and notify message listeners."""
+        message = self.STATUS_MESSAGES[value]
+        self.status = value
+        self.logger.info(value)
+        self.on_message(self, message)
+
+    @inlineCallbacks
+    def start(self):
+        self.set_status('STARTING')
+        self.logger.info("path: {}".format(self.path))
+        self.logger.info("args: {}".format(self.args))
+
+        # start listening for server connect messages
+        msg_ctx = self.client.context()
+        msg_id = msg_ctx[1]
+
+        connected = defer.Deferred()
+
+        def on_server_connect(message_ctx, msg):
+            _ID, name = msg
+            if name == self.name:
+                connected.callback(None)
+
+        message = 'Server Connect'
```

Comment:
This message gets sent by the manager, so the constant in question is in the scala code there. Could probably refactor things to put "well-known" named messages in constants somewhere, maybe in a future PR.